From e319f4edc54339f0f2d23bb308c3415a6eff51e4 Mon Sep 17 00:00:00 2001
Message-Id: <patch-15.20-e319f4edc54-20220728T160845Z-avarab@gmail.com>
In-Reply-To: <cover-00.20-00000000000-20220728T160845Z-avarab@gmail.com>
References: <cover-00.20-00000000000-20220728T160845Z-avarab@gmail.com>
From: Glen Choo <chooglen@google.com>
Date: Mon, 18 Jul 2022 12:42:40 +0200
Subject: [PATCH 15/20] submodule--helper: add "const" to copy of "update_data"
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add a "const" to the copy of "struct update_data" that's tracked by
the "struct submodule_update_clone", as it neither owns nor modifies
it.

Signed-off-by: Glen Choo <chooglen@google.com>
Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 builtin/submodule--helper.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/builtin/submodule--helper.c b/builtin/submodule--helper.c
index e235acce985..fe92c9858ce 100644
--- a/builtin/submodule--helper.c
+++ b/builtin/submodule--helper.c
@@ -1777,7 +1777,7 @@ struct submodule_update_clone {
 	int current;
 
 	/* configuration parameters which are passed on to the children */
-	struct update_data *update_data;
+	const struct update_data *update_data;
 
 	/* to be consumed by update_submodule() */
 	struct update_clone_data *update_clone;
@@ -1862,7 +1862,7 @@ static int prepare_to_clone_next_submodule(const struct cache_entry *ce,
 	const char *update_string;
 	enum submodule_update_type update_type;
 	char *key;
-	struct update_data *ud = suc->update_data;
+	const struct update_data *ud = suc->update_data;
 	char *displaypath = get_submodule_displaypath(ce->name, ud->prefix);
 	struct strbuf sb = STRBUF_INIT;
 	int needs_cloning = 0;
-- 
2.37.1.1167.g38fda70d8c4

